test/test_maskedtensor.py

```diff
@@ -96,6 +105,155 @@ def _fix_fn_name(fn_name):
     return fn_name
 
 
+class TestBasics(TestCase):
+    def test_add(self):
+        data = torch.arange(5.0)
+        mask = torch.tensor([True, True, False, True, False])
+        m0 = masked_tensor(data, mask)
+        m1 = masked_tensor(data, ~mask)
+        with self.assertRaisesRegex(ValueError, "Input masks must match."):
+            m0 + m1
+        _compare_mts(m0 + m0, masked_tensor(torch.tensor([0., 2, 0, 6, 0]), mask))
+
+    def test_softmax(self):
+        data = torch.randn(3, 4) * 0.1
+        mask = torch.tensor(
+            [
+                [True, True, True, False],
+                [False, True, False, True],
+                [True, True, False, False],
+            ]
+        )
+        mt = masked_tensor(data, mask, requires_grad=True)
+        masked_res = torch.softmax(mt, -1)
+        masked_res.sum().backward()
+        xinf = data.masked_fill(~mask, float("-inf")).detach().clone().requires_grad_()
+        tensor_res = torch.softmax(xinf, -1)
+        tensor_res.sum().backward()
+        _compare_mt_t(masked_res, tensor_res)
+        if mt.grad is not None and xinf.grad is not None:
```

Comment:
not sure I get the reason for the conditional here

Reply:
I think was an old check when the entire mask was masked out to be safe even though the entries should just be `nan`. Regardless, not necessary now for sure. Fixed